akka-http-cluster-management/src/main/scala/akka/cluster/http/management/ClusterHttpManagement.scala

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.cluster.http.management
+
+import java.util.concurrent.atomic.AtomicReference
+
+import akka.actor.AddressFromURIString
+import akka.cluster.{ Cluster, Member }
+import akka.http.scaladsl.{ ConnectionContext, Http }
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.{ Route, RouteResult }
+import akka.stream.ActorMaterializer
+import spray.json._
+
+import scala.concurrent.Future
+
+final case class ClusterUnreachableMember(node: String, observedBy: Seq[String])
+final case class ClusterMember(node: String, nodeUid: String, status: String, roles: Set[String])
+final case class ClusterMembers(selfNode: String, members: Set[ClusterMember], unreachable: Seq[ClusterUnreachableMember])
+final case class ClusterHttpManagementMessage(message: String)
+final case class ClusterHttpManagementServerInfo(hostname: String, port: Int, alreadyStarted: Boolean)
+
+sealed trait ClusterHttpManagementOperation
+case object Down extends ClusterHttpManagementOperation
+case object Leave extends ClusterHttpManagementOperation
+case object Join extends ClusterHttpManagementOperation
```

Comment:
can we make more things `private[akka]` so that we can change these without having to worry about binary compatibility?